src/librustdoc/html/render/cache.rs

```diff
@@ -373,18 +373,14 @@ crate fn get_all_types<'tcx>(
     tcx: TyCtxt<'tcx>,
 ) -> (Vec<TypeWithKind>, Vec<TypeWithKind>) {
     let mut all_types = Vec::new();
+    let mut args = Vec::new();
     for arg in decl.inputs.values.iter() {
         if arg.type_.is_self_type() {
             continue;
         }
-        // FIXME: performance wise, it'd be much better to move `args` declaration outside of the
-        // loop and replace this line with `args.clear()`.
-        let mut args = Vec::new();
         get_real_types(generics, &arg.type_, tcx, 0, &mut args);
         if !args.is_empty() {
-            // FIXME: once back to performance improvements, replace this line with:
-            // `all_types.extend(args.drain(..));`.
-            all_types.extend(args);
+            all_types.extend(args.drain(..));
```

Comment:
```suggestion
            all_types.append(&mut args);
```

This is a dedicated function for that kind of transfer and may optimize better.

Reply:
Oh indeed, thanks a lot!